source/fundamentals/builders/aggregates.txt

```diff
@@ -11,3 +11,501 @@ Aggregates Builders
+document:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin mergeOptions
+   :end-before: // end mergeOptions
+   :language: java
+   :dedent:
+
+GraphLookup
+-----------
+
+Use the ``graphLookup()`` method to create a :manual:`$graphLookup </reference/operator/aggregation/graphLookup/>`
+pipeline stage that performs a recursive search on a specified collection to match
+a specified field in one document to a specified field of another document.
+
+This example computes the social network graph for users in the
+``contacts`` collection, recursively matching the value in the ``friends`` field
+to the ``name`` field.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin graphLookupBasic
+   :end-before: // end graphLookupBasic
+   :language: java
+   :dedent:
+
+Using ``GraphLookupOptions``, you can specify the depth to recurse as well as
+the name of the depth field, if desired. In this example, ``$graphLookup`` will
+recurse up to two times, and create a field called ``degrees`` with the
+recursion depth information for every document.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin graphLookupDepth
+   :end-before: // end graphLookupDepth
+   :language: java
+   :dedent:
+
+Using ``GraphLookupOptions``, you can specify a filter that must match. In this
+example, only links with "golf" in their ``hobbies`` field will be included.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin graphLookupMatch
+   :end-before: // end graphLookupMatch
+   :language: java
+   :dedent:
+
+SortByCount
+-----------
+
+Use the ``sortByCount()`` method to create a :manual:`$sortByCount </reference/operator/aggregation/sortByCount/>`
+pipeline stage that groups documents by a given expression and then sorts
+these groups by count in descending order.
+
+.. tip::
+
+   The ``$sortByCount`` stage is identical to a ``$group`` stage with an
+   ``$sum`` accumulator followed by a ``$sort`` stage.
+
+   .. code-block:: json
+
+      [
+          { "$group": { "_id": <expression to group on>, "count": { "$sum": 1 } } },
+          { "$sort": { "count": -1 } }
+      ]
+
+This example groups documents by the truncated value of the field ``x``
+and computes the count for each distinct value.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin sortByCount
+   :end-before: // end sortByCount
+   :language: java
+   :dedent:
+
+
+ReplaceRoot
+-----------
+
+Use the ``replaceRoot()`` method to create a :manual:`$replaceRoot </reference/operator/aggregation/replaceRoot/>`
+pipeline stage that replaces each input document with the specified document.
+
+This example replaces each input document with the sub-document
+in the ``spanish_translation`` field.
```

Comment:
Question: What does the replacement document look like/what is this field getting replaced with?

Reply:
I went back and forth on larger examples, but ultimately felt they would make the page gigantic since there is so much setup. There are examples for each stage in the manual, which are linked, and we're going to add all of the simple examples from the agg book to the Aggregation page.

I'd rather keep it conceptual here for all stages to be consistent.